### ISSUE TITLE
Implement training loss and evaluation metric for Helmet datasets

### ISSUE THREAD

**Issue by mseeger** [awslabs/keys_values#72]
**Is your feature request related to a problem? Please describe.**
Several of the Helmet datasets have lists of strings as targets. We need to determine evaluation metrics for the datasets we want to use, and which training losses to use.

**Describe the solution you'd like**

- For at least some datasets (the FAQ ones): Determine which evaluation metric to use. The paper says "SubEM",
  which is 1 if one of the gold strings are in a generation of limited length, 0 otherwise
- Implement this: Must be used both to compute validation loss, and in `longcontext_eval`
- What would a training loss be in this case? Ensure this is tractable

**Comment by mseeger** [comment]
For loss: Comment on averaging over sequences in a batch:
"Token-averaged loss. We found that in the SFT stage, the distribution of the training tokens (in SFT,
the tokens from the instructions are masked out and the models are only trained on the responses) on
each GPU device can be extremely imbalanced, especially when there is synthetic data (most tokens in
a synthetic data instance are from the instruction). Conventional all-reduce loss in distributed training
averages over the sequences instead of valid tokens, which skews the optimization and also our control
over the domain proportions. Instead, we change the all-reduce loss to be the average over all valid
training tokens. (Bai et al., 2024a) implements their SFT loss in a similar way."

From https://aclanthology.org/2025.acl-long.366.pdf

==> Can do this as well!
